build/docker/Dockerfile

```diff
@@ -1,12 +1,50 @@
-FROM thawn/ttmp32gme-deps
-    
-WORKDIR /ttmp32gme
-COPY src .
+# Use modern Python 3.12 slim image as base
+FROM python:3.12-slim
+
+# Set working directory
+WORKDIR /app
+
+# Install system dependencies (wget, unzip for downloads, ffmpeg for audio conversion, git for setuptools-scm)
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        wget \
+        unzip \
+        xz-utils \
+        ca-certificates \
+        ffmpeg \
+        git && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install tttool (from releases)
+# Note: --no-check-certificate is required in some build environments due to SSL certificate chain issues.
+# This is a known issue with intermediate certificates in Docker builds.
+# For production builds with proper certificate setup, this flag can be removed.
+RUN TTTOOL_VERSION="1.8.1" && \
+    wget --no-check-certificate "https://github.com/entropia/tip-toi-reveng/releases/download/${TTTOOL_VERSION}/tttool-${TTTOOL_VERSION}.zip" && \
+    unzip -o -q "tttool-${TTTOOL_VERSION}.zip" && \
+    chmod +x tttool && \
+    mv tttool /usr/local/bin/ && \
+    rm "tttool-${TTTOOL_VERSION}.zip"
+
+# Copy application source
+COPY . /app/
+
+# Install Python dependencies
+# Note: --trusted-host flags are required in some build environments due to SSL certificate chain issues.
+# This is a known issue with intermediate certificates in Docker builds.
+# For production builds with proper certificate setup, these flags can be removed.
+RUN pip install --no-cache-dir --trusted-host pypi.org --trusted-host files.pythonhosted.org -e .
+
+# Set up directories
 ENV APPDATA=/var/lib/
-RUN mkdir config ${APPDATA}/ttmp32gme /mnt/tiptoi
+RUN mkdir -p ${APPDATA}/ttmp32gme /mnt/tiptoi
 
+# Expose port 8080
 EXPOSE 8080
 
-CMD perl ttmp32gme.pl --debug=2 --host=0.0.0.0 --port=8080 --configdir=/ttmp32gme/config
-# HEALTHCHECK --interval=5m --timeout=3s \
-  # CMD curl -f http://localhost:8080/ || exit 1
+# Run the Python backend
+CMD ["python", "-m", "ttmp32gme.ttmp32gme", "--host=0.0.0.0", "--port=8080"]
```

Comment:
Add library and config file arguments and point them to the respective file and directory in /var/lib/ttmp32gme